### PR TITLE
`TutorialScreen` implementation

### LINKED ISSUE #4: Tutorial Screen
We can add in this issue all the information and thoughts about tutorial.

### PR DESCRIPTION
**Pull request by gallottino**
Closes #4

### CODE REVIEW

**Comment by gallottino** [comment]
Not perfect yet, but you can start to review something. @Oniryu95 

Just some explanations about `tutorial.json`:
- every step of the tutorial is represented in the "flow" key of the json
- every step can have `interaction` with true/false
- if `interaction = true` there is also a list of clickable hexes
- `condition` to respect to go in the next step of the tutorial (for now just a intermediate gameGrid)

File: lib/screens/tutorial.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                    Container(
+                      decoration: const BoxDecoration(
+                          color: Colors.black,
+                          backgroundBlendMode: BlendMode
+                              .dstOut), // This one will handle background + difference out
+                    ),
+                    Align(
+                      alignment: Alignment.bottomCenter,
+                      child: AnimatedContainer(
+                        width: 650,
```
Comment (gallottino) [inline lib/screens/tutorial.dart]: Need to fix this hardcoded value

File: resources/tutorial.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
+  "gameGrid" : "- 1 - 1\n 1 2 1 -\n1 - 1 - \n- - - - ",
```
Comment (gallottino) [inline resources/tutorial.json]: "gameGrid" is set but never used in `GameLogic`

File: test/core_test.dart [hunk 3]
```diff
@@ -18,6 +18,8 @@ void main() {
       }
     }
 
+    game.getGameGridStatus();
```
Comment (gallottino) [inline test/core_test.dart]: I need to remove it

File: lib/core/game_logic.dart [hunk 4]
```diff
@@ -21,6 +21,21 @@ class GameLogic {
...
+  GameLogic.tutorial(String tutorialGrid) {
+    origin = BlockGrid.fromString(tutorialGrid);
+
+    for (int i = 0; i < origin.matrix.length; i++) {
+      for (int j = 0; j < origin.matrix[i].length; j++) {
+        origin.matrix[i][j].isEnabled = false;
+      }
+    }
+
+    origin.matrix[1][1].isEnabled = true;
```
Comment (gallottino) [inline lib/core/game_logic.dart:33]: Maybe this can be not hardcoded

**Review by Oniryu95 — APPROVED** [review]

**Pull request merged**